ci-operator/step-registry/gcp/provision/minimal-permission/gcp-provision-minimal-permission-commands.sh

```diff
@@ -26,7 +30,39 @@ fi
 # The IAM service account for IPI: ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
 # The IAM service account for UPI: upi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
 # Currently we only deal with IPI in Prow CI.
-iam_account="ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com"
 
-gcloud iam service-accounts keys create "${SHARED_DIR}/gcp_min_permissions.json" --iam-account="${iam_account}" || exit 1
-echo "$(date -u --rfc-3339=seconds) - Created a temporary key of the IAM service account for the minimum permissions testing on GCP."
+if [ -f "${CLUSTER_PROFILE_DIR}/ipi-min-permissions-sa.json" ]; then
+  echo "$(date -u --rfc-3339=seconds) - Use pre-configured key of the IAM service account for the minimum permissions testing on GCP."
+  cp "${CLUSTER_PROFILE_DIR}/ipi-min-permissions-sa.json" "${SHARED_DIR}/gcp_min_permissions.json"
+  iam_account=$(jq -r .client_email ${SHARED_DIR}/gcp_min_permissions.json)
+else
+  iam_account="ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com"
+  email=$(gcloud --project "${GOOGLE_PROJECT_ID}" iam service-accounts list --filter="email=${iam_account}" --format='value(email)')
+  if [[ -z "${email}" ]]; then
+    echo "$(date -u --rfc-3339=seconds) - Failed to find the IAM service account '${iam_account}' in GCP project '${GOOGLE_PROJECT_ID}', abort."
+    exit 1
+  fi
+  gcloud iam service-accounts keys create "${SHARED_DIR}/gcp_min_permissions.json" --iam-account="${iam_account}" || exit 1
```

Comment:
Why we restore the key creation here ? And the codes indicate that `iam_account` depends on pre-created configurations in the project.